src/components/todo/TodoList.tsx

```diff
@@ -7,11 +7,16 @@ export interface Props {
   setTodos: React.Dispatch<React.SetStateAction<TodoModal[]>>; // copied from setTodos state
 }
 
-const TodoList = ({ todos }: Props) => {
+const TodoList = ({ todos, setTodos }: Props) => {
   return (
     <div>
-      {todos.map((allTodo) => (
-        <SingleTodo todoSingleObj={allTodo} />
+      {todos.map((allTodo, index) => (
+        <SingleTodo
+          todoSingleObj={allTodo}
+          todoArrayOfObj={todos}
+          setTodos={setTodos}
+          index={index}
```

Comment:
This is bit confusing what/why we are passing all these.

todos and setTodos is familiar but not others.

Reply:
1. allTodo is storing single object of added todo items
2. index is storing the index or each todo item
We are using them while marking the tasks as done. Should I change the variable name if it is confusing, or there is any other workaround that these are not required to be passed?

Reply:
I made some modifications, kindly take a look
1. Removed allTodo, and now managing it through index only